tests/components/tedee/test_lock.py

```diff
@@ -0,0 +1,243 @@
+"""Tests for tedee lock."""
+from unittest.mock import MagicMock
+
+from pytedee_async.exception import (
+    TedeeClientException,
+    TedeeDataUpdateException,
+    TedeeLocalAuthException,
+)
+import pytest
+
+from homeassistant.components.lock import (
+    DOMAIN as LOCK_DOMAIN,
+    SERVICE_LOCK,
+    SERVICE_OPEN,
+    SERVICE_UNLOCK,
+    STATE_LOCKING,
+    STATE_UNLOCKED,
+    STATE_UNLOCKING,
+)
+from homeassistant.components.tedee.const import DOMAIN
+from homeassistant.const import ATTR_DEVICE_CLASS, ATTR_ENTITY_ID, ATTR_FRIENDLY_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, HomeAssistantError
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("init_integration")
+
+
+async def test_lock(
+    hass: HomeAssistant,
+    mock_tedee: MagicMock,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the tedee lock."""
+    mock_tedee.lock.return_value = None
+    mock_tedee.unlock.return_value = None
+    mock_tedee.open.return_value = None
+
+    state = hass.states.get("lock.lock_1a2b")
+    assert state == snapshot
+
+    entry = entity_registry.async_get(state.entity_id)
+    assert entry == snapshot
+
+    device = device_registry.async_get(entry.device_id)
+    assert device == snapshot
```

Comment:
Can you locally run 
`pytest ./tests/components/tedee --snapshot-update`

and commit the files that are generated?